src/routing/table.rs

```diff
@@ -39,25 +40,40 @@ enum FindNodeResult {
     Nodes(Vec<NodeInfo>),
 }
 
-struct RoutingTable;
+
+/// Routing table
+struct RoutingTable {
+    nodes: std::collections::HashMap,
+    node_buckets: Vec<Bucket>,
+}
 
 impl RoutingTable {
     /// Adds a node to the routing table. The node is assumed to be good until proven bad.
-    fn add_node(info: NodeInfo) {}
+    fn add_node(info: NodeInfo) {
+        let nid = info.node_id;
+        let addr = info.address;
+        nodes.insert(nid, addr);
```

Comment:
`nodes` isn't available in this scope. I think you meant to use `self.nodes`. To use self, add `&mut self` as the first parameter.

```rust
fn add_node(&mut self, info: NodeInfo)
```

This gives you mutable access to stuff in self (which you need to mutate `self.nodes`).